### ISSUE TITLE
GPL possibly?

### ISSUE THREAD

**Issue by deavmi** [anthonycr/Lightning-Browser#231]
Think about how many people would flock to a GPL browser. A license with really freedom.

**Comment by karolba** [comment]
:+1:

**Comment by deavmi** [comment]
:+1:

**Comment by kuc** [comment]
MPL 2.0 is GPL compatible unless marked "Incompatible with Secondary Licenses". And it's not marked like that, as you can see in README.md.

**Comment by karolba** [comment]
@kuc But GPL is more free. That's the point. The MIT or BSD license is also compatibile with GPL, but it doesn't guarantee that the code will not be used in proprietary software at all. MPL does that, but is not as strong as GPL.

**Comment by kuc** [comment]
@karolba Just saying. We don't know (yet) why @anthonycr chose in the past to change license from Apache 2.0 to MPL 2.0, and not to GPL (both guarantee that the code will not be used in closed-source software).

**Comment by deavmi** [comment]
GPL is the most Free though. And like me, I do not want to share my code to the furthest extent of it being in closed-source apps.

**Comment by deavmi** [comment]
The MIT license is a plain... embarrasment. Although that's opensource. GPL is Free Software.

**Comment by kuc** [comment]
However GPL is useless for licensing libraries (not the case here).

**Comment by deavmi** [comment]
Yeah, but the thing about keeping the code continuosly published is what I love.

**Comment by deavmi** [comment]
I :heart: **Free Software**.

**Comment by plumpudding** [comment]
I'd also love to be able to use a GPL browser. I agree with @deavmi about that.

**Comment by deavmi** [comment]
@plumpudding Yep. :heart:

**Comment by karolba** [comment]
Let's not forget that the Lightning Browser is only an UI for the Android WebView. Even if Lightning was licensed as GPL, the WebView still wouldn't.

**Comment by deavmi** [comment]
good point. But still it would be nice to have the UI as GPL

**Comment by deavmi** [comment]
At least it is not MIT

**Comment by anthonycr** [comment]
Should have commented sooner on this, but I had finals and stuff lol. Beware, long read below, skip to the end for a tl;dr;

I originally wanted to change from Apache to MPL because earlier on in the days of the browser, browsers started popping up to compete against Lightning that were based off Lightning Browser's own source code. I thought it was great that people liked the browser enough to use it, but it was causing a LOT of drama, and I just couldn't handle all the problems that were being caused. Browsers that originated on xda-developers like Javelin and Now were being marketed directly to users of Lightning, and I didn't have the time or energy to deal with the drama that surrounded these small browsers when they didn't abide by the license. I had better things to do than defend my open-source project (like actually build stuff), so I decided to put a barrier that would discourage these drama-filled, primarily-for-profit forks, and encourage derivatives that were open source (allowing features to be brought back to Lightning) but still allow larger companies to license the work properly.

I'm no lawyer, so I originally chose MPL because it strongly discouraged the straight up copying that the apache license let go on, and it might have helped because I've seen a drop in the adware clones. I could have gone GPL due to its similarities, but after some research, I came to the understanding that companies shy away from GPL projects because of the strong language and requirement that every part of a GPL project be open-sourced, and that isn't realistic for a company wishing to add in a proprietary blob that has their API keys or something. For example, Hola VPN uses an old version of Lightning in their source, but they wouldn't have been able to use it if it was GPL because it would expose their whole framework and undermine their paid product. Additionally, I wasn't certain how well the GPL played with the ability to give a company a paid license for proprietary use. Since the goal of the license is not to drive away large commercial use, but simply discourage improper use of the license and protect the project from cannibalistic competition, I didn't want to put such a strong shield on the project.

MPL seemed like a nice intermediate license, so I went with it. Those are my main reasons, idk what will be gained at this point by switching to GPL other than to have the name of a more popular license stamped on the repo as some sort of advertising ¯\_(ツ)_/¯

Of course, my reasons for MPL may be completely compatible with GPL, so if my reasoning is wrong, please correct me. GPL would probably encourage more contribution to the repo, since it is more permanently free, but I'm making from this project, so I don't want to encourage wide-spread contribution from people who might not be cool with me making money on it.

All that said, I'm working for a company starting soon where I will be working on an android browser, so if I cannot continue working on Lightning due to contractual issues, making Lightning completely free and changing the license to GPL is probably a good idea to keep the project alive. From what I've read/signed I am able to continue work on it as long as I don't do obviously nefarious things like use code I wrote for work, since the browser's IP was conceived prior to my employment, but I have to be prepared for the possibility in the future. As much as it would be dope to work on this project full time, it's not going to supply enough income for me to justify not getting a job at a big company.

tl;dr: as long as I'm actively working on the project, it makes sense to me to keep it MPL, but I may not be able to work on it as I enter the industry as a software engineer, in which case changing the project to GPL will give the project a higher likelihood of living on without me.

**Comment by anthonycr** [comment]
Sorry for rambling on, I got caught up in writing that and have a hard time condensing my thoughts.

**Comment by kuc** [comment]
@anthonycr 

>Additionally, I wasn't certain how well the GPL played with the ability to give a company a paid license for proprietary use.

As author, you can always double-license, it's not something uncommon - example: http://ckeditor.com/about/license
Only thing you can't do is "revoking" license retroactively of already released code.

**Comment by kuc** [comment]
>All that said, I'm working for a company starting soon where I will be working on an android browser, so if I cannot continue working on Lightning due to contractual issues

I don't think it's even legal to do if they want to do this.

>From what I've read/signed I am able to continue work on it as long as I don't do obviously nefarious things like use code I wrote for work, since the browser's IP was conceived prior to my employment

That's standard rule, however some companies release part of their inner libraries as open-source, so then you could do this.

**Comment by deavmi** [comment]
@kuc Yeah, companies cannot control what you do when you are coding something lese not business related. You could work at apple, but make an OS under the GPL. They can't do anything.

From my understanding.

That sounds wrong, it's a human right to **work on what you want**.

ANyway we all have opinions, I am a GNU Boy. So yeah. GPL for thw in, cheers. :heart: :smile:

**Comment by deavmi** [comment]
Hence the difference between open-source and Free Software.

Open-source: Charity work that gets disposed. (Unless revoken by licensor)

Free Software: Charity work - Infinite Lifespan. (Unless revoken by licensor)
